cmd/influx/secret.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/cmd/influx/internal"
+	"github.com/influxdata/influxdb/http"
+	"github.com/spf13/cobra"
+	input "github.com/tcnksm/go-input"
+)
+
+type secretSVCsFn func() (influxdb.SecretService, influxdb.OrganizationService, func(*input.UI) string, error)
+
+func cmdSecret(opts ...genericCLIOptFn) *cobra.Command {
+	return newCmdSecretBuilder(newSecretSVCs, opts...).cmd()
+}
+
+type cmdSecretBuilder struct {
+	genericCLIOpts
+
+	svcFn secretSVCsFn
+
+	key string
+	org organization
+}
+
+func newCmdSecretBuilder(svcsFn secretSVCsFn, opts ...genericCLIOptFn) *cmdSecretBuilder {
+	opt := genericCLIOpts{
+		in: os.Stdin,
+		w:  os.Stdout,
+	}
+	for _, o := range opts {
+		o(&opt)
+	}
+
+	return &cmdSecretBuilder{
+		genericCLIOpts: opt,
+		svcFn:          svcsFn,
+	}
+}
+
+func (b *cmdSecretBuilder) cmd() *cobra.Command {
+	cmd := b.newCmd("secret", nil)
+	cmd.Short = "Secret management commands"
+	cmd.Run = seeHelp
+	cmd.AddCommand(
+		b.cmdDelete(),
+		b.cmdFind(),
+		b.cmdUpdate(),
+	)
+	return cmd
+}
+
+func (b *cmdSecretBuilder) cmdUpdate() *cobra.Command {
+	cmd := b.newCmd("update", b.cmdUpdateRunEFn)
+	cmd.Short = "Update secret"
+	cmd.Flags().StringVarP(&b.key, "key", "k", "", "The secret key (required)")
+	cmd.MarkFlagRequired("key")
+	b.org.register(cmd, false)
+
+	return cmd
+}
+
+func (b *cmdSecretBuilder) cmdDelete() *cobra.Command {
+	cmd := b.newCmd("delete", b.cmdDeleteRunEFn)
+	cmd.Short = "Delete secret"
+
+	cmd.Flags().StringVarP(&b.key, "key", "k", "", "The secret key (required)")
+	cmd.MarkFlagRequired("key")
+	b.org.register(cmd, false)
+
+	return cmd
+}
+
+func (b *cmdSecretBuilder) cmdUpdateRunEFn(cmd *cobra.Command, args []string) error {
+	scrSVC, orgSVC, getSecretFn, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+	orgID, err := b.org.getID(orgSVC)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	ui := &input.UI{
+		Writer: b.genericCLIOpts.w,
+		Reader: b.genericCLIOpts.in,
+	}
+	secret := getSecretFn(ui)
+
+	if err := scrSVC.PatchSecrets(ctx, orgID, map[string]string{
```

Comment:
nit: can we pull out all multi calls out of the if block?

```go
err := scrSVC.PatchSecrets(ctx, orgID, map[string]string{
...
})
if err != nil {
```

when it grows it turns into a frankenstein like this 😢:
https://github.com/influxdata/influxdb/blob/master/cmd/influxd/launcher/launcher.go#L664

Reply:
sure